client/src/bastionlab/polars/remote_polars.py

```diff
@@ -313,64 +315,161 @@ def join_asof(
             ),
         )
 
-    def histplot(self: LDF, col_x: str, col_y: str, bins: int = 10, **kwargs):
+    def histplot(
+        self: LDF, x: str = "count", y: str = "count", bins: int = 10, **kwargs
+    ):
+        """Histplot plots a univariate histogram, where one x or y axes is provided or a bivariate histogram, where both x and y axes values are supplied.
+
+        Histplot filters down a RemoteLazyFrame to necessary columns only, groups x axes into bins
+        and performs aggregated queries before calling either Seaborn's barplot (for univaritate histograms) or heatmap function (for bivariate histograms),
+        which helps us to limit data retrieved from the server to a minimum.
+        Args:
+            x (str): The name of column to be used for x axes. Default value is "count", which trigger pl.count() to be used on this axes.
+            y (str): The name of column to be used for y axes. Default value is "count", which trigger pl.count() to be used on this axes.
+            bins (int): An integer bin value which x axes will be grouped by. Default value is 10.
+            **kwargs: Other keyword arguments that will be passed to Seaborn's barplot function, in the case of one column being supplied, or heatmap function, where both x and y columns are supplied.
+        """
+
+        col_x = x if x != None else "count"
+        col_y = y if y != None else "count"
+
+        if col_x == "count" and col_y == "count":
+            print("Please provide an 'x' or 'y' value")
+            return
+
         model = ApplyBins(bins)
-        df = (
-            self.filter(pl.col(col_x) != None)
-            .filter(pl.col(col_y) != None)
-            .select([pl.col(col_y), pl.col(col_x)])
-            .apply_udf([col_x], model)
-            .groupby(pl.col(col_x))
-            .agg(pl.col(col_y).count())
-            .sort(col_x)
-            .collect()
-            .fetch()
-            .to_pandas()
-        )
-        sns.barplot(x=df[col_x], y=df[col_y], **kwargs)
+
+        # if we have only X or Y
+        if col_x == "count" or col_y == "count":
+            q_x = pl.col(col_x) if col_x != "count" else pl.col(col_y)
+            q_y = pl.count()
+
+            if not col_x in self.columns and not col_y in self.columns:
+                print("Error: column not found in dataframe")
```

Comment:
Why not raising an exception here? This way the user can catch it and have a clear idea of what went wrong.